### ISSUE TITLE
filestore_utils: unify operation function signature format

### ISSUE THREAD

**Issue by zchcai** [google/fuzzbench#397]
For example, in `common/gsutil.py`, we have two formats of function signatures for those file operations, similar to:

https://github.com/google/fuzzbench/blob/185fb114f7fe4b2419a8f445c0186cad23145692/common/gsutil.py#L30-L31

And

https://github.com/google/fuzzbench/blob/185fb114f7fe4b2419a8f445c0186cad23145692/common/gsutil.py#L62-L72

How about we use one unified format? I prefer the `rsync` one.

We only need to think about, what will be the general options? For example, `recursive` or `-r`.

**Comment by lszekeres** [comment]
Yes, whenever possible we should prefer explicit parameters (like `source`, `destination`, `recursive`) as opposed to *args and **kwargs. E.g., `cp` should have and explicit `source` and `destination`, I agree with you. This helps both readability and using these util functions.

We can fix the the current signatures based on the current use cases, and if needed add more explicit params gradually based on new use cases.

**Comment by zchcai** [comment]
Also, this issue also includes a second task: the removal attempt for `gsutil_options` in signature.
Here is the PR related in my fork: [remove gsutil_options #7](https://github.com/zchcai/fuzzbench/pull/7)

**Comment by jonathanmetzman** [comment]
I guess I'm in favor of this if it works. I may have done *cp_arguments arguments intentionally to support uses where I want to add in some option without having to support it explicitly like I do in rsync. If we don't need to support explicitly then fine to remove. If we do, then I think we should do whatever is cleanest (ie: if making all the arguments explicit is too ugly then we shouldn't do it).

**Comment by zchcai** [comment]
I think we can provide some common options explicitly for every file system. And leave others to `options` with specific keys, to indicate that, when using a typical file storage, we want to add some options.

I prefer the `options["<file_system_type>"]` format. This is neat and easy for scaling.

**Comment by zchcai** [comment]
Another benefit for specifying some common options is to get the file storage src/dest path easily. Then this `filestore_utils` will be independent, not relying on `get_cloud_experiment_path`.
@jonathanmetzman  we can also have `options`, which will complete what you want as `*_argument`.

**Comment by zchcai** [comment]
This issue may not be an issue any more.
The starting of this issue is we might need `source` and `dest` to decide which file storage to use. Now we deny this design after a discussion. See https://github.com/google/fuzzbench/issues/398 and https://github.com/google/fuzzbench/pull/394 for details.

**Comment by zchcai** [comment]
After #402 and #405 merged, we can open an PR for this issue.
@jonathanmetzman @lszekeres
